torchdynamo/utils.py

```diff
@@ -51,6 +56,25 @@ def time_wrapper(*args, **kwargs):
         compilation_metrics[key].append(time.time() - t0)
         return r
 
+    def profile_wrapper(*args, **kwargs):
+        global timer_counter
+        datafn = (
+            func.__name__ + f"{next(timer_counter)}.profile"
+        )  # Name the data file sensibly
+        prof = cProfile.Profile()
+        prof.enable()
```

Comment:
will this work correctly if @dynamo_timed is applied on a parent function and a child function, and thus re-enters cprofile?  i'm not sure if the current usage of dynamo_timed avoids this reentrancy but it could easily happen.